doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,280 @@
+# Advanced Analytics Framework User Guide
+
+The Advanced Analytics Framework (AAF) enables implementing complex data analysis algorithms with Exasol. Users can use the features of AAF in their custom implementations.
+
+## Table of Contents
+
+* [Setup](#setup)
+* [Usage](#usage)
+* [Custom Algorithms](#custom-algorithms)
+
+## Setup
+
+### Exasol database
+
+* The Exasol cluster must already be running with version 7.1 or later.
+* Database connection information and credentials are needed for the database itself as also for the BucketFS.
+
+### BucketFS Connection
+
+AAF employs some Lua scripts and User Defined Functions (UDFs).  The Lua scripts are orchestrating the UDFs while the UDFs are performing the actual analytic functions.
+
+AAF keeps a common state of execution and passes input data and results between Lua and UDFs via files in the Bucket File System (BucketFS) of the Exasol database.
```

Comment:
This is basically a minimalistic explanation, maybe it should go into its own section

Reply:
See next push.